lib/rules/RuleSetCompiler.js

```diff
@@ -343,7 +343,9 @@ class RuleSetCompiler {
 	 */
 	error(path, value, message) {
 		return new Error(
-			`Compiling RuleSet failed: ${message} (at ${path}: ${value})`
+			`Compiling RuleSet failed: ${message} (at ${path}: ${JSON.stringify(
```

Comment:
Could you try to use `util.inspect` for this? It should be able to handle circular data, limit the depth, show regexps and display the data in a nicer way.

But afaik the formating depends on the node.js version so test case might be a bit more involved...